Source/Diagnostics/ReducedDiags/FieldReduced.cpp

```diff
@@ -0,0 +1,258 @@
+        {
+            // open file
+            std::ofstream ofs;
+            ofs.open(m_path + m_rd_name + "." + m_extension,
+                std::ofstream::out | std::ofstream::app);
+            // write header row
+            ofs << "#";
+            ofs << "[1]step()";
+            ofs << m_sep;
+            ofs << "[2]time(s)";
+
+            if (m_fieldEnergy)
+            {
+                for (int lev = 0; lev < nLevel; ++lev)
+                {
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(3+3*lev) + "]";
+                    ofs << "total_lev"+std::to_string(lev)+"(J)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(4+3*lev) + "]";
+                    ofs << "E_lev"+std::to_string(lev)+"(J)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(5+3*lev) + "]";
+                    ofs << "B_lev"+std::to_string(lev)+"(J)";
+                }
+            }
+
+            if (m_maxField)
+            {
+                for (int lev = 0; lev < nLevel; ++lev)
+                {
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+3+8*lev) + "]";
+                    ofs << "max_Ex_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+4+8*lev) + "]";
+                    ofs << "max_Ey_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+5+8*lev) + "]";
+                    ofs << "max_Ez_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+6+8*lev) + "]";
+                    ofs << "max_|E|_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+7+8*lev) + "]";
+                    ofs << "max_Bx_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+8+8*lev) + "]";
+                    ofs << "max_By_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+9+8*lev) + "]";
+                    ofs << "max_Bz_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+10+8*lev) + "]";
+                    ofs << "max_|B|_lev"+std::to_string(lev)+" (T)";
+                }
+            }
+
+            ofs << std::endl;
+            // close file
+            ofs.close();
+        }
+    }
+
+}
+// end constructor
+
+// function that computes field energy and/or maximum field values
+void FieldReduced::ComputeDiags (int step)
+{
+
+    // Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
```

Comment:
Can this use the new interval syntax, too? :)

Reply:
Yes, but I guess we'd have to implement this for all reduced diagnostics at the same time. (`m_freq` is an attribute of the `ReducedDiags` class) Maybe it's better to do this in a separate PR for clarity?